### ISSUE TITLE
The riverpod_generator package does not support ChangeNotifier values

### ISSUE THREAD

**Issue by agordeev** [lucavenir/go_router_riverpod#17]
In `router.dart`:

<img width="474" alt="Screenshot 2023-03-11 at 15 12 54" src="https://user-images.githubusercontent.com/2883524/224483738-216ed7db-3ead-4f64-9c7f-305dab67f920.png">

This warning comes from `riverpod_lint`. The app works fine though.

**Comment by lucavenir** [comment]
Hi! These examples were written before the linter was implemented. I'd have to review this lint warning, but it looks like a false positive to me. I'll look up into it and I'll let you know.

**Comment by lucavenir** [comment]
Hi! I've talked with Remi about this just to be sure.

It is not a false positive and - alas - the linter is right. I didn't think about it, but `go_router` _is_ a `ChangeNotifier`. See its definition:
```dart
class GoRouter extends ChangeNotifier implements RouterConfig<RouteMatchList> {
  ...
}
```

Thus, what the linter is trying is:
> look out, if you inject this provider you _will not_ get any updates (no rebuilds) because the generator doesn't support `ChangeNotifier`

Which is fine for this usage. In the end, you _never_ want to directly inject a `GoRouter` instance around your code. That's why `GoRouter` APIs have a dedicated `InheritedWidget` (and its relative extension methods, e.g. `context.go`).

Nonetheless, I'll change the code and _remove_ this at least in the codegen version. I don't want to spread bad practices AND honestly it doesn't really matter much; as I said above, `router` isn't injected anywhere else.

Thank you for this feedback, as you've probably helped tons of other devs with this potential issue.
Closing and fixing ASAP.
